app/core/models/ocelot/stanzas/CalculationStanza.scala

```diff
@@ -125,16 +158,54 @@ sealed trait Operation {
 
 case class AddOperation(left: String, right: String, label: String) extends Operation {
 
-  def eval(labels: Labels): Labels = op(_ + _, (s1:String, s2:String) => Some(s1 + s2), unsupportedOperation("Add"), labels)
+  def eval(labels: Labels): Labels = {
 
+    val updatedLabels: Labels = operands(labels) match {
+      case (Some(x), None, Some(y), None) => op(x, y, _ + _, (s1:String, s2:String) => Some(s1 + s2), unsupportedOperation("Add"), labels)
+      case (None, Some(xList), Some(y), None) => listAndValueOp(xList, y, appendStringToList, labels)
+      case (Some(x), None, None, Some(yList)) => listAndValueOp(yList, x, prependStringToList, labels)
+      case (None, Some(xList), None, Some(yList)) => listOp(xList, yList, addListToList, labels)
+      case _ => unsupportedOperation("Add")(None, None)
+        labels
+    }
+
+    updatedLabels
+  }
+
+  private def appendStringToList(l: List[String], s: String): Option[List[String]] = Some((s :: l.reverse).reverse)
+
+  private def prependStringToList(l: List[String], s: String): Option[List[String]] = Some(s :: l)
+
+  private def addListToList(list1: List[String], list2: List[String]): List[String] = list1 ::: list2
 }
 
 case class SubtractOperation(left: String, right: String, label: String) extends Operation {
 
-  def eval(labels: Labels): Labels = op(_ - _, unsupportedOperation("Subtract"), subtractDate, labels)
+  def eval(labels: Labels): Labels = {
+
+    operands(labels) match {
+      case (Some(x), None, Some(y), None) => op(x, y, _ - _, unsupportedOperation("Subtract"), subtractDate, labels)
+      case (None, Some(xList), Some(y), None) => listAndValueOp(xList, y, subtractStringFromList, labels)
+      case (Some(x), None, None, Some(yList)) => listAndValueOp(yList, x, unsupportedListOperation("Subtract list from string"), labels)
+      case (None, Some(xList), None, Some(yList)) => listOp(xList, yList, subtractListFromList, labels)
+      case _ => unsupportedOperation("Subtract")(None, None)
+        labels
+    }
+  }
 
   private def subtractDate(date: LocalDate, other: LocalDate) : Option[String] =
     Some(other.until(date, ChronoUnit.DAYS).toString)
+
+  private def subtractStringFromList(l: List[String], s: String): Option[List[String]] = Some(l.filter(_ != s))
+
+  @tailrec
+  private def subtractListFromList(list1: List[String], list2: List[String]): List[String] = {
```

Comment:
{}  not required on subtractListFromList()

Reply:
Done